### DISCUSSION TITLE
I have encountered an urgent issue about perfermance.

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8997] [General]
I'm using fabric v6.0.1 now, and I created a new class which is similar to Group, some codes as follows:
`export class Container extends Group {
  title: any;
  _objectsByGroup: any[];
  constructor(objects?: any, options?: any, objectsRelativeToGroup?: boolean) {
    super(objects, options, objectsRelativeToGroup)
  }
 addIntoCanvas(object: any): any {
    this.canvas.add(object);   
    this.addByRelatived(object)
  }
  addByRelatived(object: any) {
    object._container = this
    if (!object) return
    //这里是为了避免重复添加，用在切换容器时需要做判断
    //标题不放在_objects对象集合上
    if (this._objects.indexOf(object) < 0) {
      this._objects.push(object)
    }
    //要加true否则会有缓存
    this._onObjectAdded(object)
    //为了让子元素的lineCoords显示正常，否则最开始的时候无法选中一级容器的子元素
    object.setCoords()
  }
`
In this class which is called Container, you can add child into Container by those codes(the method addIntoCanvas) : 
1. canvas.add(object), //put the child into canvas directly;
2. _onObjectAdded(object), //building a relationship through the method _onObjectAdded from Group.

maybe you will ask why don't you use the method add from Group directly, I tried, but I've encountered many questions.

Now I want to ask a question about rendering performance, is there a big performance difference between the method 'addIntoCanvas' and the method add from Group, if I want to render about 200 hundreds children elements at the same time.

For example:
A：for(let i =0;i<200;i++){
addIntoCanvas(object);
}
B：for(let i =0;i<200;i++){
add(object);
}

I want to know the performance differences between A and B.

**Comment by ShaMan123** [comment]
If caching is on the the group will render all your objects once onto the cache.
Then if nothing changes each time it renders it will pull the cache canvas and render that, which is an image, so it is performant.
Try switching group objectCaching to false and that should make perf degrade to the same level as what you are reporting

**Reply by wuxingping** [reply]
Thank you very much,
I want to make sure if I got what you mean, so as you said there will be no diffenerce between A and B if I do like this :

A:
`let container = new Container([],{top:0,left:0, objectCaching:false,});`
`let rect = new Rect({width:100,height:100,fill:"red"});`
`container.addIntoCanvas(rect );`
the important point is that I set objectCaching to false.

B:
`let group = new Group([],{top:0,left:0});`
`let rect = new Rect({width:100,height:100,fill:"red"});`
`group .add(rect );`

Am I right?

**Reply by ShaMan123** [reply]
The other way around.
Since you are adding to canvas from the container (I would not do that) then objectCaching has no effect because nothing is rendered by the container.
```diff
B:
- let group = new Group([],{top:0,left:0});
+ let group = new Group([],{top:0,left:0, objectCaching: false });
```

**Reply by wuxingping** [reply]
ok,  I see.
